Fancy demo scene panels

sandera0606 opened the pull request:
# Fancy Demo Scene
https://www.notion.so/uwrl/Fancy-Demo-Scene-11ebc072402f81649d0ce9bf4d6887a7?pvs=4

### Summary of Changes
- Created recording panel prefab
     - Recording, twohand/onehand toggles, recording list (read from jsondata folder), recording deletion, recording preview all work
     - Not sure if the recording list can actually scroll because I'm not sure how to test that 
- Created welcome panel prefab
- Modified JsonWriter script to put JsonWrite() function into a coroutine so the recording counter isn't slowed down
- Changed some JsonWriter access modifiers to private but added new get/set methods when appropriate
- Added UIButton script
- created some new prefabs for enemies
- created some new UIButton prefabs just for convenience
- updated recording UI script 
- Added new "Control In Editor" script for user to control recording duration, gesture name, (and an empty space for file path, but I didn't do that and it might not be necessary)
    - made a "Configure Recording" prefab to house that script
- made new subfolder in Assets for UI components
- Imported new font

### Screenshots of Changes to Scenes/Prefabs
Changes to Scenes: None

Prefabs:
#### Configure Recording
![image](https://github.com/user-attachments/assets/a30ca8f5-30f7-475f-ab07-a0891a24d405)


#### Recording Status UI
![image](https://github.com/user-attachments/assets/c78577fe-0fab-4360-a7a9-089f64fa9dec)


#### Welcome Screen
![image](https://github.com/user-attachments/assets/7cd6ef53-c93b-44a4-973f-005e90f7966d)


#### Json Writer (fewer public variables)
![image](https://github.com/user-attachments/assets/b16a84f2-1be4-412c-99a0-207a7fc64006)

sandera0606 committed:
recording panel, face raiders

added recording panel + face raiders asset

sandera0606 committed:
adding sparks designs

did welcome screen, middle of working on recording screen

sandera0606 committed:
Updated recording status UI

Things still not working:
- actual deletion of files
    - the x button on the recordings doesn't work yet either but idk why
- allow user choose folder to save json files in

sandera0606 committed:
recording screen done!

did everything except i can't test the scrollyness

sandera0606 committed:
made configure recording prefab

JustinLin905 committed:
Merge branch 'main' into fancy-demo-scene-panels

JustinLin905 commented on Assets/Scripts/JsonWriter.cs:
```diff
@@ -179,4 +206,19 @@ public void SetRecordingHandModeTwoHands()
     {
         recordingHandMode = HandMode.TwoHands;
     }
```
Oops, since you changed recordingHandMode to be private in this class, that's causing a compilation error because TestRequest.cs is trying to directly access it. You need to make a public GetRecordingHandMode() function that'll return the value so other scripts can use it. You can add it underneath these two setter methods.

sandera0606 committed:
fixed bugs

added GetRecordingHandMode() function, dragged JsonWriter into the Recording Status UI gameobject in the main scene

sandera0606 commented:
Bug fixed!

GetRecordingHandMode() function added in JsonWriter

Also made a small change in the main scene - dragged Scripts>JsonWriter into a serialized field in the RecordingStatusUI gameobject

sandera0606 committed:
Added gesture testing panel
